hs_core/models.py

```diff
@@ -2751,6 +2751,10 @@ def size(self):
     def modified_time(self):
         return self.resource_file.storage.get_modified_time(self.resource_file.name)
 
+    @property
+    def checksum(self):
+        return self.resource_file.storage.checksum(self.resource_file.name)
```

Comment:
As currently implemented, the checksum method in iRODS Storage class recomputes checksum each time, which would give us a performance hit especially for large files if this checksum property is called often. The reason to use -f flag is that the file could be updated, and we don't want to return the staled checksum. I actually used extra_data field to store checksum for published resource bag, so that it can be retrieved directly in multiple places without calling this irods storage checksum method to avoid checksum recomputation. This works well for published resource bag since published resource bag checksums never change again, but I think it will require some thinking-through here for a generic checksum get property for any file in a resource that could get updated.

Reply:
Seems like we could just add a field to the ResourceFile model and set the checksum field when the file is uploaded.

Reply:
or, implement another checksum method without the -f.  Do you have a preference @hyi?